curvefs/src/metaserver/inode_manager.cpp

```diff
@@ -0,0 +1,218 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: 2021-05-19
+ * Author: chenwei
+ */
+#include <glog/logging.h>
+#include "curvefs/src/metaserver/inode_manager.h"
+#include "curvefs/src/common/define.h"
+#include "src/common/timeutility.h"
+
+namespace curvefs {
+namespace metaserver {
+MetaStatusCode InodeManager::CreateInode(uint32_t fsId, uint64_t length,
+                                uint32_t uid, uint32_t gid, uint32_t mode,
+                                FsFileType type, std::string symlink,
+                                Inode *newInode) {
+    LOG(INFO) << "CreateInode, fsId = " << fsId
+              << ", length = " << length
+              << ", uid = " << uid
+              << ", gid = " << gid
+              << ", mode = " << mode
+              << ", type =" << FsFileType_Name(type)
+              << ", symlink = " << symlink;
+    if (type == FsFileType::TYPE_SYM_LINK
+        && symlink.empty()) {
+        return MetaStatusCode::SYM_LINK_EMPTY;
+    }
+
+    // 1. generate inode
+    Inode inode;
+    inode.set_inodeid(GetNextId());
+    inode.set_fsid(fsId);
+    inode.set_length(length);
+    inode.set_uid(uid);
+    inode.set_gid(gid);
+    inode.set_mode(mode);
+    inode.set_type(type);
+    inode.set_mtime(curve::common::TimeUtility::GetTimeofDayMs());
+    inode.set_atime(curve::common::TimeUtility::GetTimeofDayMs());
+    inode.set_ctime(curve::common::TimeUtility::GetTimeofDayMs());
```

Comment:
I think it's better to local a variable for `GetTimeofDayMs()`, and then set relative time for inode, because the `mtime`, `atime`, `ctime` maybe not same in current implementation.

Reply:
For create situation, the mtime, atime, ctime is the same.